[SECFEAT-1484] Adds Twilio Sign up section (#6093)
segmentio/segment-docs: src/segment-app/iam/index.md

With Twilio Unified Login, Twilio users can use their Twilio email, password, and authentication settings to access several Twilio products, including Segment. Once you link your Segment account to your Twilio credentials, you can access Segment directly from the Twilio console using the [Twilio Product Switcher](#twilio-product-switcher).

### Twilio Sign up

> Segment sign ups and invitations that are re-directed to Twilio's sign up page adheres to [Twilio's minimum password requirements and 2FA](https://help.twilio.com/articles/115012261968). The intention is to provide a single experience for users to access various Twilio products, including Twilio Messaging, SendGrid, and Segment. You can also use other providers, such as Google sign up. To learn more, review Twilio's [Account Management](https://support.twilio.com/hc/en-us/sections/205104908-Account-Management?_gl=1*1xa50pg*_ga*OTUyMjQ4OTU5LjE2NjM2ODQzMDE.*_ga_3JKYB4GBBY*MTcwNzc2ODE1OC4xNDkuMS4xNzA3NzY4MjUzLjAuMC4w) documentation.

Reviewer: ```suggestion
Segment invitations and sign ups that are redirected to Twilio's sign up page must adhere to Twilio's [minimum password and 2FA requirements](https://help.twilio.com/articles/115012261968){:target="_blank”}. You can also use [Sign in With Google](https://developers.google.com/identity/gsi/web/guides/overview){:target="_blank”} to sign in to your Twilio account. To learn more, view Twilio's [Account Management](https://support.twilio.com/hc/en-us/sections/205104908-Account-Management?_gl=1*1xa50pg*_ga*OTUyMjQ4OTU5LjE2NjM2ODQzMDE.*_ga_3JKYB4GBBY*MTcwNzc2ODE1OC4xNDkuMS4xNzA3NzY4MjUzLjAuMC4w){:target="_blank”} documentation.
```

Author: The copy on the auth0 button for signing up with Google is `Sign Up with Google`. That said, we adhere to this text and prefer not to include the overview page to prevent confusion with the end user because of this.

<img width="976" alt="Screenshot 2024-02-20 at 2 55 17 PM" src="https://github.com/segmentio/segment-docs/assets/178258/3f58ba7e-5c0b-4d3e-ad19-ff655d0733e1">

Reviewer: okay that makes sense! thanks

> Any existing Segment user will adhere to existing password requirements and 2FA settings at the Workspace level.

Reviewer: ```suggestion
Any existing Segment user must adhere to existing password requirements and 2FA settings set at the Workspace level.
```
